File: src/main/java/com/gabrielczar/semantic/dto/LocationCityDTO.java
package com.gabrielczar.semantic.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.gabrielczar.semantic.utils.GeometryUtil;
import com.vividsolutions.jts.geom.Polygon;
import lombok.Data;

@Data
public class LocationCityDTO {
    private LocationDTO initial;

    @JsonProperty("final")
    private LocationDTO ending;

    private String city;

    public Polygon getGeometry() {
        return GeometryUtil.createPolygon(
                initial.getLatitude(),
                initial.getLongitude(),
                ending.getLatitude(),
                ending.getLongitude()
        );
    }
}

File: src/main/java/com/gabrielczar/semantic/dto/LocationCityDTO.kt
package com.gabrielczar.semantic.dto

import com.fasterxml.jackson.annotation.JsonProperty
import com.gabrielczar.semantic.utils.createPolygon
import com.vividsolutions.jts.geom.Polygon

data class LocationCityDTO (
        val initial : LocationDTO,

        @JsonProperty("final")
        val ending : LocationDTO,

        var city : String

){
    fun getGeometry() : Polygon {
        return createPolygon(
                initial.latitude,
                initial.longitude,
                ending.latitude,
                ending.longitude
        )
    }
}